lib/serviceClient.ts

```diff
@@ -151,8 +161,13 @@ export interface ServiceClientOptions {
  */
 export class ServiceClient {
   /**
-   * If specified, this is the base URI that requests will be made against for this ServiceClient.
-   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   * If specified:
+   * - This `baseUri` becomes the base URI that requests will be made against for this ServiceClient.
+   * - If a `TokenCredential` was passed through the constructor, this `baseUri` defines the `getToken` scope to be `${options.baseUri}/.default`.
+   *
+   * If it is not specified:
+   * - All OperationSpecs must contain a baseUrl property.
+   * - If a `TokenCredential` was passed through the constructor, the `getToken` scope is set to be "https://management.azure.com/.default".
```

Comment:
Rewording to call out that baseUri set after the client is constructed does not get used by the adapter

```suggestion
   * The base URI against which requests will be made when using this ServiceClient instance.
   *
   * This can be set either by setting the `baseUri` in the `options` parameter to the ServiceClient constructor or directly after constructing the ServiceClient.
   * If set via the ServiceClient constructor when using the overload that takes the `TokenCredential`, this base URI sets the scope used to get the AAD token to `${baseUri}/.default` instead of the default "https://management.azure.com/.default"
   *
   * If it is not specified, all OperationSpecs must contain a baseUrl property.
```